api/admin/controller.py

```diff
@@ -852,29 +852,16 @@ def custom_lists(self):
 
             old_entries = list.entries
             for entry in entries:
-                identifier, ignore = Identifier.parse_urn(self._db, entry.get("id"))
-                pools = self._db.query(
-                    LicensePool
-                ).join(
-                    LicensePool.collection
-                ).join(
-                    LicensePool.identifier
-                ).join(
-                    Collection.libraries
-                ).filter(
-                    Identifier.id==identifier.id
-                ).filter(
-                    Library.id==library.id
-                ).all()
-                if not pools:
-                    return CANNOT_ADD_CUSTOM_LIST_ENTRY_WITHOUT_LICENSEPOOL
-
-                edition = pools[0].presentation_edition
-                list.add_entry(edition, featured=True)
-
-            new_urns = [entry.get("id") for entry in entries]
+                pwid = entry.get("pwid")
+                edition = get_one(self._db, Edition,
+                                  on_multiple='interchangeable',
+                                  permanent_work_id=pwid)
+                if edition and edition.work:
+                    list.add_entry(edition.work, featured=True)
```

Comment:
This code could be a little nicer if you brought back the joins and got one Work that has an Edition with the given permanent work ID.